### PR TITLE
Adjustments for upstream upgrades

### PR DESCRIPTION
**Pull request by dothinking**
Some adjustments to handle the side effects of upgrading `PyMuPDF` from 1.23.8 to 1.23.14+, e.g.,

- behavior change of `fitz.Rect()` -> adjusted from `pdf2docx` side. #250
- behavior issue of `fitz.Pixmap()`  -> downgrade `pymupdf` for now.  https://github.com/pymupdf/PyMuPDF/issues/3058

And some other tweaks, e.g., remove Github Action for publishing docs to Github page.

### CODE REVIEW

**Comment by JorjMcKie** [comment]
A fix is identified already and will lead to a new PyMuPDF version  - hopefully even today.

**Comment by jamie-lemon** [comment]
Thanks Train! ( hope you don't me calling you "Train" - seen you use it a few times, please let me know ) . It is possible that we will have fixed some of the issues you are seeing in an imminent release of PyMuPDF.

Adding Harald & Julian here as they should be aware of the behaviour change for `Rect` & `Pixmap` - I think we know about the `Pixmap` one already. I guess if the issues are both fixed in PyMuPDF 1.23.16 then you may want to rework this PR.

**Comment by dothinking** [comment]
@jamie-lemon "Train" sounds friendly for me, as it's a name used in my college time. 😁

Regarding this PR, once released PyMuPDF 1.23.16, I just need to delete the version restriction in requirements.txt, the rest should be OK.

**Review by JorjMcKie — APPROVED** [review]

File: pdf2docx/common/Element.py [hunk 1]
```diff
@@ -172,43 +178,45 @@ def get_main_bbox(self, e, threshold:float=0.95):
...
+        i.e. perpendicular to reading direction.
+
         Args:
             e (Element): Object to check with
-            factor (float, optional): Threshold of overlap ratio, the larger it is, the higher probability the two bbox-es are aligned.
-            text_direction (bool, optional): Consider text direction or not. True by default, from left to right if False.
+            factor (float, optional): Threshold of overlap ratio, the larger it is, the higher
+                probability the two bbox-es are aligned.
+            text_direction (bool, optional): Consider text direction or not.
+                True by default,from left to right if False.
```
Comment (jamie-lemon) [inline pdf2docx/common/Element.py:202]: I don't really understand what this means "True by default, from left to right if False".

So if the `text_direction` is `True` then we do consider text direction - okay, but if it is `False` then we don't consider text directions, however when I read this it seems like `False` means we consider a left to right text direction as it says "from left to right if False". I'm confused!

File: pdf2docx/common/Element.py [hunk 2]
```diff
@@ -225,29 +233,32 @@ def vertically_align_with(self, e, factor:float=0.0, text_direction:bool=True):
...
+        i.e. along the reading direction.
+
         Args:
             e (Element): Element to check with
-            factor (float, optional): threshold of overlap ratio, the larger it is, the higher probability the two bbox-es are aligned.
-            text_direction (bool, optional): consider text direction or not. True by default, from left to right if False.
+            factor (float, optional): threshold of overlap ratio, the larger it is, the higher
+                probability the two bbox-es are aligned.
+            text_direction (bool, optional): consider text direction or not.
+                True by default, from left to right if False.
```
Comment (jamie-lemon) [inline pdf2docx/common/Element.py:244]: See previous comment.

**Review by jamie-lemon — CHANGES_REQUESTED** [review]
Just didn't understand one of the comments.

**Comment by julian-smith-artifex-com** [comment]
FYI, PyMuPDF-1.23.16 has just been released, and fixes the Pixmap issue https://github.com/pymupdf/PyMuPDF/issues/3058.

The fitz.Rect problem was introduced in 1.23.9 and fixed in 1.23.13 and later.

Hope that helps.

File: pdf2docx/common/Element.py (hunk 1 above)
Comment (dothinking) [reply pdf2docx/common/Element.py:202]: I didn't read any contradictions between your thoughts -> don't consider text directions -> ignore real text directions -> use default text directions -> the most common case, horizontal, i.e., from left to right.

Appreciated if you help a precise wording.

Comment (dothinking) [reply pdf2docx/common/Element.py:202]: Per my understanding, don't consider text directions, means to use default text direction, which is from left to right.

Comment (jamie-lemon) [reply pdf2docx/common/Element.py:202]: I suppose the bit I don't understand then is if `True` then what is the text direction? Basically:

`False` = from left to right
`True` = ?

Also if we are `False` then we do consider the text direction don't we (left to right)? Which is why `text_direction (bool, optional): Consider text direction or not.` confuses me!

Comment (dothinking) [reply pdf2docx/common/Element.py:202]: I probably understand your confusion. When `False`, the words `from left to right` does not mean to text direction, but the default direction for "horizontal". I should just keep `True by default` and delete the rest.

image

Comment (jamie-lemon) [reply pdf2docx/common/Element.py:202]: Okay - yes please just delete to avoid the confusion.

**Review by jamie-lemon — APPROVED** [review]

**Pull request merged**